src/plugins/timelist/Timelist.vue

```diff
@@ -163,6 +180,57 @@ export default {
                 this.listActivities();
             }
         },
+        addItem(domainObject) {
+            this.planObjects = [domainObject];
+            this.resetPlanData();
+            if (domainObject.type === 'plan') {
+                this.getPlanDataAndSetConfig({
+                    ...this.domainObject,
+                    selectFile: domainObject.selectFile
+                });
+            }
+        },
+        addToComposition(telemetryObject) {
+            if (this.planObjects.length > 0) {
+                this.confirmRemoval(telemetryObject);
+            } else {
+                this.addItem(telemetryObject);
+            }
+        },
+        confirmRemoval(telemetryObject) {
+            const dialog = this.openmct.overlays.dialog({
+                iconClass: 'alert',
+                message: 'This action will replace the current plan. Do you want to continue?',
+                buttons: [
+                    {
+                        label: 'Ok',
+                        emphasis: true,
+                        callback: () => {
+                            const oldTelemetryObject = this.planObjects[0];
+                            this.removeFromComposition(oldTelemetryObject);
+                            this.addItem(telemetryObject);
+                            dialog.dismiss();
+                        }
+                    },
+                    {
+                        label: 'Cancel',
+                        callback: () => {
+                            this.removeFromComposition(telemetryObject);
+                            dialog.dismiss();
+                        }
+                    }
+                ]
+            });
+        },
+        removeFromComposition(telemetryObject) {
+            this.composition.remove(telemetryObject);
+        },
+        removeItem() {
```

Comment:
I think you could get rid of this, it's only used in the comp listener, you could just replace it with resetPlanData?

Reply:
You think you're sooooo smart! :P

Reply:
haha, definitely DO NOT think that :)